<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scDesign3">
<title>Evaluate clustering goodness-of-fit by scDesign3 â€¢ scDesign3</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Evaluate clustering goodness-of-fit by scDesign3">
<meta property="og:description" content="scDesign3">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scDesign3</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.5</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/scDesign3.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/scDesign3-introduction-vignette.html">scDesign3 Introduction</a>
    <a class="dropdown-item" href="../articles/scDesign3-marginal-vignette.html">scDesign3 marginal distribution for genes</a>
    <a class="dropdown-item" href="../articles/scDesign3-copulaCompare-vignette.html">Compare gaussian copula and vine copula</a>
    <a class="dropdown-item" href="../articles/scDesign3-librarySize-vignette.html">Simulate datasets with cell library size</a>
    <a class="dropdown-item" href="../articles/scDesign3-multipleLineages-vignette.html">Simulate datasets with multiple lineages</a>
    <a class="dropdown-item" href="../articles/scDesign3-conditionEffect-vignette.html">Simulate datasets with condition effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-batchEffect-vignette.html">Simulate datasets with batch effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-spatial-vignette.html">Simulate spatial transcriptomic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-spatial-deconvolution.html">Simulate spot-resolution spatial data for cell-type deconvolution</a>
    <a class="dropdown-item" href="../articles/scDesign3-scATACseq-vignette.html">Simulate single-cell ATAC-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-CITEseq-vignette.html">Simulate CITE-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-multiomics-vignette.html">Simulate multi-omics data from single-omic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-clusterGOF-vignette.html">Evaluate clustering goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-pseudotimeGOF-vignette.html">Evaluate pseudotime goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-DEanalysis-vignette.html">Benchmarking DE Analysis with scDesign3</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/SONGDONGYUAN1994/scDesign3/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">

<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Evaluate clustering goodness-of-fit by scDesign3</h1>
                        <h4 data-toc-skip class="author">Dongyuan Song</h4>
            <address class="author_afil">
      Bioinformatics IDP, University of California, Los Angeles<br><a class="author_email" href="mailto:#"></a><a href="mailto:dongyuansong@ucla.edu" class="email">dongyuansong@ucla.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Qingyang Wang</h4>
            <address class="author_afil">
      Department of Statistics, University of California, Los Angeles<br><a class="author_email" href="mailto:#"></a><a href="mailto:qw802@g.ucla.edu" class="email">qw802@g.ucla.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">15 July 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/SONGDONGYUAN1994/scDesign3/blob/HEAD/../../scDesign3/code/vignettes/scDesign3-clusterGOF-vignette.Rmd" class="external-link"><code>../../scDesign3/code/vignettes/scDesign3-clusterGOF-vignette.Rmd</code></a></small>
      <div class="d-none name"><code>scDesign3-clusterGOF-vignette.Rmd</code></div>
    </div>

    
    
<style type="text/css">
pre {
  white-space: pre !important;
  overflow-x: scroll !important;
}
</style>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/SONGDONGYUAN1994/scDesign3" class="external-link">scDesign3</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DuoClustering2018</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jchiquet/aricode" class="external-link">aricode</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In this tutorial, we will show how to use scDesign3 to evaluate the clustering goodness-of-fit for different cell-type assignments. If the true labels are unavailable and we have little prior knowledge, the scDesign3 BIC can serve as an unsupervised metric.</p>
</div>
<div class="section level2">
<h2 id="read-in-the-reference-data">Read in the reference data<a class="anchor" aria-label="anchor" href="#read-in-the-reference-data"></a>
</h2>
<p>The raw data is from the R package <code>DuoClustering2018</code> which contain a set of datasets with various clustering results.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Zhengmix4eq_sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html" class="external-link">get</a></span><span class="op">(</span><span class="st">"sce_filteredExpr10_Zhengmix4eq"</span><span class="op">)</span><span class="op">(</span>metadata <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html" class="external-link">get</a></span><span class="op">(</span><span class="st">"clustering_summary_filteredExpr10_Zhengmix4eq_v1"</span><span class="op">)</span><span class="op">(</span>metadata <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>  </span>
<span><span class="va">res_sub</span> <span class="op">&lt;-</span> <span class="va">res</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">method</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SC3"</span>, <span class="st">"Seurat"</span>, <span class="st">"PCAHC"</span>, <span class="st">"PCAKmeans"</span>, <span class="st">"CIDR"</span><span class="op">)</span> <span class="op">&amp;</span> <span class="va">run</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">res_sub_list</span> <span class="op">&lt;-</span> <span class="va">res_sub</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">dataset</span>, <span class="va">method</span>, <span class="va">run</span>, <span class="va">k</span>, <span class="va">resolution</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_split.html" class="external-link">group_split</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>For demonstration purpose, we use the <code>Zhengmix4eq</code> dataset in the package with top 100 highly variable genes and the corresponding k-means clustering results with k = <span class="math inline">\(2 ,\cdots, 10\)</span>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kmeans_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/funprog.html" class="external-link">Filter</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">method</span> <span class="op">==</span> <span class="st">"PCAKmeans"</span><span class="op">)</span><span class="op">}</span>, <span class="va">res_sub_list</span><span class="op">)</span></span>
<span><span class="va">ncell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">Zhengmix4eq_sce</span><span class="op">)</span></span>
<span><span class="va">ngene</span> <span class="op">&lt;-</span> <span class="fl">100</span></span>
<span><span class="va">zheng_sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVar.html" class="external-link">modelGeneVar</a></span><span class="op">(</span><span class="va">Zhengmix4eq_sce</span><span class="op">)</span></span>
<span><span class="va">chosen</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/getTopHVGs.html" class="external-link">getTopHVGs</a></span><span class="op">(</span><span class="va">zheng_sce</span>, n <span class="op">=</span> <span class="va">ngene</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">Zhengmix4eq_sce</span><span class="op">[</span><span class="va">chosen</span>, <span class="op">]</span></span>
<span><span class="va">ntrain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">ncell</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span></span>
<span>  </span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">train_index</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">ncell</span><span class="op">)</span>, <span class="va">ntrain</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">train_sce</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span>, <span class="va">train_index</span><span class="op">]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="simulation">Simulation<a class="anchor" aria-label="anchor" href="#simulation"></a>
</h2>
<p>We then use different cell-type clustering information to simulate new data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">scDesign3_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">kmeans_res</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="va">dat</span> <span class="op">&lt;-</span> <span class="va">x</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cell"</span>, <span class="st">"cluster"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">dat</span><span class="op">$</span><span class="va">cell</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">train_sce</span><span class="op">)</span><span class="op">$</span><span class="va">cell_type</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">dat</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">train_sce</span><span class="op">)</span>, <span class="op">]</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span></span>
<span>      <span class="va">simu_sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scdesign3.html">scdesign3</a></span><span class="op">(</span>sce <span class="op">=</span> <span class="va">train_sce</span>, </span>
<span>                            celltype <span class="op">=</span> <span class="st">'cell_type'</span>, </span>
<span>                            pseudotime <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                            spatial <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                            other_covariates <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                            corr_formula <span class="op">=</span> <span class="st">"1"</span>, </span>
<span>                            mu_formula <span class="op">=</span> <span class="st">"cell_type"</span>, </span>
<span>                            sigma_formula <span class="op">=</span> <span class="st">"cell_type"</span>, </span>
<span>                            n_cores <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                            copula <span class="op">=</span> <span class="st">"gaussian"</span>, </span>
<span>                            assay_use <span class="op">=</span> <span class="st">"counts"</span>, </span>
<span>                            family_use <span class="op">=</span> <span class="st">"nb"</span><span class="op">)</span></span>
<span>      </span>
<span>      <span class="va">simu_sce</span></span>
<span>    <span class="op">}</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<p>After the simulations, we can check the BIC provided by our package and the calculated ARI to evaluate k-means clustering qualities.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bic_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">scDesign3_result</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">model_bic</span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="va">kmeans_ari</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">kmeans_res</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/pkg/aricode/man/ARI.html" class="external-link">ARI</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">cluster</span>, <span class="va">x</span><span class="op">$</span><span class="va">trueclass</span><span class="op">)</span><span class="op">}</span><span class="op">)</span></span>
<span><span class="va">bic_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">bic_list</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">bic_list</span><span class="op">)</span>, byrow <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">bic_df</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">bic_list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">bic_df</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"k = "</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">metric</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>ari <span class="op">=</span> <span class="va">kmeans_ari</span>, bic <span class="op">=</span> <span class="va">bic_df</span><span class="op">$</span><span class="va">bic.marginal</span>, Method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"k = "</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p_cluster_metric</span> <span class="op">&lt;-</span> <span class="va">metric</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span><span class="va">ari</span>, y <span class="op">=</span> <span class="va">bic</span>,label <span class="op">=</span> <span class="va">Method</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    panel.grid.minor <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.grid.major <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggpubr</span><span class="fu">::</span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/stat_cor.html" class="external-link">stat_cor</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"spearman"</span>, cor.coef.name <span class="op">=</span> <span class="st">"rho"</span>, label.x.npc <span class="op">=</span> <span class="st">"left"</span>, label.y.npc <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"scDesign3 BIC"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"ARI"</span><span class="op">)</span></span>
<span><span class="va">p_cluster_metric</span></span></code></pre></div>
<p><img src="scDesign3-clusterGOF-vignette_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 20.04.6 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/Los_Angeles</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] ggplot2_3.4.2               aricode_1.0.2              </span></span>
<span><span class="co">#&gt;  [3] scran_1.28.1                scuttle_1.10.1             </span></span>
<span><span class="co">#&gt;  [5] SingleCellExperiment_1.22.0 SummarizedExperiment_1.30.2</span></span>
<span><span class="co">#&gt;  [7] Biobase_2.60.0              GenomicRanges_1.52.0       </span></span>
<span><span class="co">#&gt;  [9] GenomeInfoDb_1.36.1         IRanges_2.34.1             </span></span>
<span><span class="co">#&gt; [11] S4Vectors_0.38.1            BiocGenerics_0.46.0        </span></span>
<span><span class="co">#&gt; [13] MatrixGenerics_1.12.2       matrixStats_1.0.0          </span></span>
<span><span class="co">#&gt; [15] dplyr_1.1.2                 DuoClustering2018_1.18.0   </span></span>
<span><span class="co">#&gt; [17] scDesign3_0.99.5            BiocStyle_2.28.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] jsonlite_1.8.7                magrittr_2.0.3               </span></span>
<span><span class="co">#&gt;   [3] farver_2.1.1                  rmarkdown_2.23               </span></span>
<span><span class="co">#&gt;   [5] fs_1.6.2                      zlibbioc_1.46.0              </span></span>
<span><span class="co">#&gt;   [7] ragg_1.2.5                    vctrs_0.6.3                  </span></span>
<span><span class="co">#&gt;   [9] memoise_2.0.1                 DelayedMatrixStats_1.22.1    </span></span>
<span><span class="co">#&gt;  [11] RCurl_1.98-1.12               rstatix_0.7.2                </span></span>
<span><span class="co">#&gt;  [13] htmltools_0.5.5               S4Arrays_1.0.4               </span></span>
<span><span class="co">#&gt;  [15] AnnotationHub_3.8.0           curl_5.0.1                   </span></span>
<span><span class="co">#&gt;  [17] broom_1.0.5                   BiocNeighbors_1.18.0         </span></span>
<span><span class="co">#&gt;  [19] sass_0.4.6                    bslib_0.5.0                  </span></span>
<span><span class="co">#&gt;  [21] desc_1.4.2                    plyr_1.8.8                   </span></span>
<span><span class="co">#&gt;  [23] cachem_1.0.8                  igraph_1.5.0                 </span></span>
<span><span class="co">#&gt;  [25] mime_0.12                     lifecycle_1.0.3              </span></span>
<span><span class="co">#&gt;  [27] pkgconfig_2.0.3               rsvd_1.0.5                   </span></span>
<span><span class="co">#&gt;  [29] Matrix_1.6-0                  R6_2.5.1                     </span></span>
<span><span class="co">#&gt;  [31] fastmap_1.1.1                 GenomeInfoDbData_1.2.10      </span></span>
<span><span class="co">#&gt;  [33] shiny_1.7.4.1                 digest_0.6.33                </span></span>
<span><span class="co">#&gt;  [35] colorspace_2.1-0              AnnotationDbi_1.62.2         </span></span>
<span><span class="co">#&gt;  [37] rprojroot_2.0.3               dqrng_0.3.0                  </span></span>
<span><span class="co">#&gt;  [39] irlba_2.3.5.1                 ExperimentHub_2.8.0          </span></span>
<span><span class="co">#&gt;  [41] textshaping_0.3.6             RSQLite_2.3.1                </span></span>
<span><span class="co">#&gt;  [43] ggpubr_0.6.0                  beachmat_2.16.0              </span></span>
<span><span class="co">#&gt;  [45] labeling_0.4.2                filelock_1.0.2               </span></span>
<span><span class="co">#&gt;  [47] fansi_1.0.4                   abind_1.4-5                  </span></span>
<span><span class="co">#&gt;  [49] mgcv_1.8-42                   httr_1.4.6                   </span></span>
<span><span class="co">#&gt;  [51] compiler_4.3.0                bit64_4.0.5                  </span></span>
<span><span class="co">#&gt;  [53] withr_2.5.0                   backports_1.4.1              </span></span>
<span><span class="co">#&gt;  [55] BiocParallel_1.34.2           carData_3.0-5                </span></span>
<span><span class="co">#&gt;  [57] viridis_0.6.3                 DBI_1.1.3                    </span></span>
<span><span class="co">#&gt;  [59] highr_0.10                    ggsignif_0.6.4               </span></span>
<span><span class="co">#&gt;  [61] MASS_7.3-60                   rappdirs_0.3.3               </span></span>
<span><span class="co">#&gt;  [63] DelayedArray_0.26.6           bluster_1.10.0               </span></span>
<span><span class="co">#&gt;  [65] tools_4.3.0                   interactiveDisplayBase_1.38.0</span></span>
<span><span class="co">#&gt;  [67] httpuv_1.6.11                 glue_1.6.2                   </span></span>
<span><span class="co">#&gt;  [69] nlme_3.1-162                  promises_1.2.0.1             </span></span>
<span><span class="co">#&gt;  [71] grid_4.3.0                    cluster_2.1.4                </span></span>
<span><span class="co">#&gt;  [73] reshape2_1.4.4                generics_0.1.3               </span></span>
<span><span class="co">#&gt;  [75] gtable_0.3.3                  tidyr_1.3.0                  </span></span>
<span><span class="co">#&gt;  [77] car_3.1-2                     metapod_1.8.0                </span></span>
<span><span class="co">#&gt;  [79] BiocSingular_1.16.0           ScaledMatrix_1.8.1           </span></span>
<span><span class="co">#&gt;  [81] utf8_1.2.3                    XVector_0.40.0               </span></span>
<span><span class="co">#&gt;  [83] BiocVersion_3.17.1            pillar_1.9.0                 </span></span>
<span><span class="co">#&gt;  [85] stringr_1.5.0                 limma_3.56.2                 </span></span>
<span><span class="co">#&gt;  [87] later_1.3.1                   splines_4.3.0                </span></span>
<span><span class="co">#&gt;  [89] BiocFileCache_2.8.0           lattice_0.21-8               </span></span>
<span><span class="co">#&gt;  [91] survival_3.5-5                bit_4.0.5                    </span></span>
<span><span class="co">#&gt;  [93] gamlss.data_6.0-2             tidyselect_1.2.0             </span></span>
<span><span class="co">#&gt;  [95] locfit_1.5-9.8                Biostrings_2.68.1            </span></span>
<span><span class="co">#&gt;  [97] knitr_1.43                    gridExtra_2.3                </span></span>
<span><span class="co">#&gt;  [99] bookdown_0.34                 edgeR_3.42.4                 </span></span>
<span><span class="co">#&gt; [101] xfun_0.39                     statmod_1.5.0                </span></span>
<span><span class="co">#&gt; [103] stringi_1.7.12                yaml_2.3.7                   </span></span>
<span><span class="co">#&gt; [105] evaluate_0.21                 codetools_0.2-19             </span></span>
<span><span class="co">#&gt; [107] tibble_3.2.1                  BiocManager_1.30.21          </span></span>
<span><span class="co">#&gt; [109] cli_3.6.1                     xtable_1.8-4                 </span></span>
<span><span class="co">#&gt; [111] systemfonts_1.0.4             munsell_0.5.0                </span></span>
<span><span class="co">#&gt; [113] jquerylib_0.1.4               Rcpp_1.0.11                  </span></span>
<span><span class="co">#&gt; [115] dbplyr_2.3.3                  png_0.1-8                    </span></span>
<span><span class="co">#&gt; [117] parallel_4.3.0                ellipsis_0.3.2               </span></span>
<span><span class="co">#&gt; [119] pkgdown_2.0.7                 blob_1.2.4                   </span></span>
<span><span class="co">#&gt; [121] mclust_6.0.0                  sparseMatrixStats_1.12.2     </span></span>
<span><span class="co">#&gt; [123] bitops_1.0-7                  gamlss.dist_6.0-5            </span></span>
<span><span class="co">#&gt; [125] mvtnorm_1.2-2                 viridisLite_0.4.2            </span></span>
<span><span class="co">#&gt; [127] ggthemes_4.2.4                scales_1.2.1                 </span></span>
<span><span class="co">#&gt; [129] gamlss_5.4-12                 purrr_1.0.1                  </span></span>
<span><span class="co">#&gt; [131] crayon_1.5.2                  rlang_1.1.1                  </span></span>
<span><span class="co">#&gt; [133] KEGGREST_1.40.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Dongyuan Song, Qingyang Wang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
