<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scDesign3">
<title>Simulate multi-omics data from single-omic data • scDesign3</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Simulate multi-omics data from single-omic data">
<meta property="og:description" content="scDesign3">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scDesign3</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.5</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/scDesign3.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/scDesign3-introduction-vignette.html">scDesign3 Introduction</a>
    <a class="dropdown-item" href="../articles/scDesign3-marginal-vignette.html">scDesign3 marginal distribution for genes</a>
    <a class="dropdown-item" href="../articles/scDesign3-copulaCompare-vignette.html">Compare gaussian copula and vine copula</a>
    <a class="dropdown-item" href="../articles/scDesign3-librarySize-vignette.html">Simulate datasets with cell library size</a>
    <a class="dropdown-item" href="../articles/scDesign3-multipleLineages-vignette.html">Simulate datasets with multiple lineages</a>
    <a class="dropdown-item" href="../articles/scDesign3-conditionEffect-vignette.html">Simulate datasets with condition effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-batchEffect-vignette.html">Simulate datasets with batch effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-spatial-vignette.html">Simulate spatial transcriptomic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-spatial-deconvolution.html">Simulate spot-resolution spatial data for cell-type deconvolution</a>
    <a class="dropdown-item" href="../articles/scDesign3-scATACseq-vignette.html">Simulate single-cell ATAC-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-CITEseq-vignette.html">Simulate CITE-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-multiomics-vignette.html">Simulate multi-omics data from single-omic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-clusterGOF-vignette.html">Evaluate clustering goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-pseudotimeGOF-vignette.html">Evaluate pseudotime goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-DEanalysis-vignette.html">Benchmarking DE Analysis with scDesign3</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/SONGDONGYUAN1994/scDesign3/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">

<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Simulate multi-omics data from single-omic data</h1>
                        <h4 data-toc-skip class="author">Dongyuan Song</h4>
            <address class="author_afil">
      Bioinformatics IDP, University of California, Los Angeles<br><a class="author_email" href="mailto:#"></a><a href="mailto:dongyuansong@ucla.edu" class="email">dongyuansong@ucla.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Qingyang Wang</h4>
            <address class="author_afil">
      Department of Statistics, University of California, Los Angeles<br><a class="author_email" href="mailto:#"></a><a href="mailto:qw802@g.ucla.edu" class="email">qw802@g.ucla.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">17 July 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/SONGDONGYUAN1994/scDesign3/blob/HEAD/../../scDesign3/code/vignettes/scDesign3-multiomics-vignette.Rmd" class="external-link"><code>../../scDesign3/code/vignettes/scDesign3-multiomics-vignette.Rmd</code></a></small>
      <div class="d-none name"><code>scDesign3-multiomics-vignette.Rmd</code></div>
    </div>

    
    
<style type="text/css">
pre {
  white-space: pre !important;
  overflow-x: scroll !important;
}
</style>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/SONGDONGYUAN1994/scDesign3" class="external-link">scDesign3</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/teunbrand/ggh4x" class="external-link">ggh4x</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/tkonopka/umap" class="external-link">umap</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In this tutorial, we will show how to use scDesign3 to simulate multi-omics (RNA expression + DNA methylation) data by learning from real data that only have a single modality. The example data and aligned low-dimensional embeddings are from <a href="https://github.com/caokai1073/Pamona" class="external-link">Pamona</a>.</p>
</div>
<div class="section level2">
<h2 id="read-in-the-reference-data">Read in the reference data<a class="anchor" aria-label="anchor" href="#read-in-the-reference-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">SCGEMMETH_sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://www.dropbox.com/sh/0tptrft7hsstpi3/AAAo-79sEfTkvrEig0q4L5Oia/SCGEMMETH_sce.rds?raw=1"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">SCGEMRNA_sce</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://www.dropbox.com/sh/0tptrft7hsstpi3/AADEKgmZG4FsW5tdbu1fdYKda/SCGEMRNA_sce.rds?raw=1"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">SCGEMMETH_sce</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">SCGEMRNA_sce</span><span class="op">)</span></span></code></pre></div>
<p>We first combine the cell-level data from the scRNA-seq data and the DNA methylation data.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">coldata_bind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">SCGEMRNA_sce</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Tech <span class="op">=</span> <span class="st">"RNA"</span><span class="op">)</span>, <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">SCGEMMETH_sce</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Tech <span class="op">=</span> <span class="st">"Methylation"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">coldata_bind</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="simulation">Simulation<a class="anchor" aria-label="anchor" href="#simulation"></a>
</h2>
<p>We first use the step-by-step functions to fit genes’ marginal models, copulas, and extract simulation parameters separately for the scRNA-seq data and the DNA methylation data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">RNA_data</span> <span class="op">&lt;-</span> <span class="fu">scDesign3</span><span class="fu">::</span><span class="fu"><a href="../reference/construct_data.html">construct_data</a></span><span class="op">(</span></span>
<span>  <span class="va">SCGEMRNA_sce</span>, </span>
<span>  assay_use <span class="op">=</span> <span class="st">"logcounts"</span>, </span>
<span>  celltype <span class="op">=</span> <span class="st">"cell_type"</span>, </span>
<span>  pseudotime <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>  spatial <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMAP1_integrated"</span>, <span class="st">"UMAP2_integrated"</span><span class="op">)</span>, </span>
<span>  other_covariates <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>  corr_by <span class="op">=</span> <span class="st">"1"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">METH_data</span> <span class="op">&lt;-</span> <span class="fu">scDesign3</span><span class="fu">::</span><span class="fu"><a href="../reference/construct_data.html">construct_data</a></span><span class="op">(</span></span>
<span>  <span class="va">SCGEMMETH_sce</span>, </span>
<span>  assay_use <span class="op">=</span> <span class="st">"counts"</span>, </span>
<span>  celltype <span class="op">=</span> <span class="st">"cell_type"</span>, </span>
<span>  pseudotime <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>  spatial <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMAP1_integrated"</span>, <span class="st">"UMAP2_integrated"</span><span class="op">)</span>, </span>
<span>  other_covariates <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>  corr_by <span class="op">=</span> <span class="st">"1"</span><span class="op">)</span></span></code></pre></div>
<p>Note here we actually treat the 2D aligned UMAPs as a kind of “pseudo”-spatial data. We use the tensor regression spline to fit two ref datasets seperately.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">RNA_marginal</span> <span class="op">&lt;-</span> <span class="fu">scDesign3</span><span class="fu">::</span><span class="fu"><a href="../reference/fit_marginal.html">fit_marginal</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">RNA_data</span>, </span>
<span>  predictor <span class="op">=</span> <span class="st">"gene"</span>, </span>
<span>  mu_formula <span class="op">=</span> <span class="st">"te(UMAP1_integrated, UMAP2_integrated, bs = 'cr', k = 10)"</span>, </span>
<span>  sigma_formula <span class="op">=</span> <span class="st">"te(UMAP1_integrated, UMAP2_integrated, bs = 'cr', k = 5)"</span>, </span>
<span>  family_use <span class="op">=</span> <span class="st">"gaussian"</span>, </span>
<span>  n_cores <span class="op">=</span> <span class="fl">1</span>, </span>
<span>  usebam <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">METH_marginal</span> <span class="op">&lt;-</span> <span class="fu">scDesign3</span><span class="fu">::</span><span class="fu"><a href="../reference/fit_marginal.html">fit_marginal</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">METH_data</span>, </span>
<span>  predictor <span class="op">=</span> <span class="st">"gene"</span>, </span>
<span>  mu_formula <span class="op">=</span> <span class="st">"te(UMAP1_integrated, UMAP2_integrated, bs = 'cr', k = 10)"</span>, </span>
<span>  sigma_formula <span class="op">=</span> <span class="st">"1"</span>, </span>
<span>  family_use <span class="op">=</span> <span class="st">"binomial"</span>, </span>
<span>  n_cores <span class="op">=</span> <span class="fl">1</span>, </span>
<span>  usebam <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">RNA_copula</span> <span class="op">&lt;-</span> <span class="fu">scDesign3</span><span class="fu">::</span><span class="fu"><a href="../reference/fit_copula.html">fit_copula</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">SCGEMRNA_sce</span>,</span>
<span>    assay_use <span class="op">=</span> <span class="st">"logcounts"</span>,</span>
<span>    marginal_list <span class="op">=</span> <span class="va">RNA_marginal</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>    copula <span class="op">=</span> <span class="st">"vine"</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    input_data <span class="op">=</span> <span class="va">RNA_data</span><span class="op">$</span><span class="va">dat</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">METH_copula</span> <span class="op">&lt;-</span> <span class="fu">scDesign3</span><span class="fu">::</span><span class="fu"><a href="../reference/fit_copula.html">fit_copula</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">SCGEMMETH_sce</span>,</span>
<span>    assay_use <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>    marginal_list <span class="op">=</span> <span class="va">METH_marginal</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"binomial"</span>,</span>
<span>    copula <span class="op">=</span> <span class="st">"vine"</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    input_data <span class="op">=</span> <span class="va">METH_data</span><span class="op">$</span><span class="va">dat</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">RNA_para</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_para.html">extract_para</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">SCGEMRNA_sce</span>,</span>
<span>    assay_use <span class="op">=</span> <span class="st">"logcounts"</span>,</span>
<span>    marginal_list <span class="op">=</span> <span class="va">RNA_marginal</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>    new_covariate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">RNA_data</span><span class="op">$</span><span class="va">dat</span>, <span class="va">METH_data</span><span class="op">$</span><span class="va">dat</span><span class="op">)</span>,</span>
<span>    data <span class="op">=</span> <span class="va">RNA_data</span><span class="op">$</span><span class="va">dat</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">METH_para</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_para.html">extract_para</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">SCGEMMETH_sce</span>,</span>
<span>    marginal_list <span class="op">=</span> <span class="va">METH_marginal</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"binomial"</span>,</span>
<span>    new_covariate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">RNA_data</span><span class="op">$</span><span class="va">dat</span>, <span class="va">METH_data</span><span class="op">$</span><span class="va">dat</span><span class="op">)</span>,</span>
<span>    data <span class="op">=</span> <span class="va">METH_data</span><span class="op">$</span><span class="va">dat</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="simulate-new-datasets">Simulate New Datasets<a class="anchor" aria-label="anchor" href="#simulate-new-datasets"></a>
</h3>
<p>Then, we combined the cell covariates from both the scRNA-seq data and the DNA methylation data as the new covariate to simulate the two new datasets with parameters from scRNA-seq data and the DNA methylation data separately.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">RNA_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simu_new.html">simu_new</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">SCGEMRNA_sce</span>,</span>
<span>    assay_use <span class="op">=</span> <span class="st">"logcounts"</span>,</span>
<span>    mean_mat <span class="op">=</span> <span class="va">RNA_para</span><span class="op">$</span><span class="va">mean_mat</span>,</span>
<span>    sigma_mat <span class="op">=</span> <span class="va">RNA_para</span><span class="op">$</span><span class="va">sigma_mat</span>,</span>
<span>    zero_mat <span class="op">=</span> <span class="va">RNA_para</span><span class="op">$</span><span class="va">zero_mat</span>,</span>
<span>    quantile_mat <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    copula_list <span class="op">=</span> <span class="va">RNA_copula</span><span class="op">$</span><span class="va">copula_list</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>    input_data <span class="op">=</span> <span class="va">RNA_data</span><span class="op">$</span><span class="va">dat</span>,</span>
<span>    new_covariate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">RNA_data</span><span class="op">$</span><span class="va">dat</span>, <span class="va">METH_data</span><span class="op">$</span><span class="va">dat</span><span class="op">)</span>,</span>
<span>    important_feature <span class="op">=</span> <span class="va">RNA_copula</span><span class="op">$</span><span class="va">important_feature</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">METH_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simu_new.html">simu_new</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">SCGEMMETH_sce</span>,</span>
<span>    mean_mat <span class="op">=</span> <span class="va">METH_para</span><span class="op">$</span><span class="va">mean_mat</span>,</span>
<span>    sigma_mat <span class="op">=</span> <span class="va">METH_para</span><span class="op">$</span><span class="va">sigma_mat</span>,</span>
<span>    zero_mat <span class="op">=</span> <span class="va">METH_para</span><span class="op">$</span><span class="va">zero_mat</span>,</span>
<span>    quantile_mat <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    copula_list <span class="op">=</span> <span class="va">METH_copula</span><span class="op">$</span><span class="va">copula_list</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"binomial"</span>,</span>
<span>    input_data <span class="op">=</span> <span class="va">METH_data</span><span class="op">$</span><span class="va">dat</span>,</span>
<span>    new_covariate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">RNA_data</span><span class="op">$</span><span class="va">dat</span>, <span class="va">METH_data</span><span class="op">$</span><span class="va">dat</span><span class="op">)</span>,</span>
<span>    important_feature <span class="op">=</span> <span class="va">METH_copula</span><span class="op">$</span><span class="va">important_feature</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<p>We combine the two synthetic datasets and obtain the UMAP embeddings for the combined dataset.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">count_combine</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">RNA_res</span>, <span class="va">METH_res</span><span class="op">)</span></span>
<span><span class="va">count_combine_pca</span> <span class="op">&lt;-</span> <span class="fu">irlba</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/irlba/man/prcomp_irlba.html" class="external-link">prcomp_irlba</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">count_combine</span><span class="op">)</span>, <span class="fl">5</span>, scale. <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">count_combine_umap</span> <span class="op">&lt;-</span> <span class="fu">umap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/umap/man/umap.html" class="external-link">umap</a></span><span class="op">(</span><span class="va">count_combine_pca</span><span class="op">$</span><span class="va">x</span>, n_neighbors<span class="op">=</span><span class="fl">30</span>, min_dist<span class="op">=</span><span class="fl">0.7</span><span class="op">)</span><span class="op">$</span><span class="va">layout</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">count_combine_umap</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMAP1"</span>, <span class="st">"UMAP2"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">SCGEMNEW_sce</span> <span class="op">&lt;-</span> <span class="fu">SingleCellExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>logcounts <span class="op">=</span> <span class="va">count_combine</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDims</a></span><span class="op">(</span><span class="va">SCGEMNEW_sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>PCA <span class="op">=</span> <span class="va">count_combine_pca</span><span class="op">$</span><span class="va">x</span>, UMAP <span class="op">=</span> <span class="va">count_combine_umap</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">SCGEMRNA_umap</span> <span class="op">&lt;-</span> <span class="fu">umap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/umap/man/umap.html" class="external-link">umap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">SCGEMRNA_sce</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"X"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span>, n_neighbors<span class="op">=</span><span class="fl">30</span>, min_dist<span class="op">=</span><span class="fl">0.7</span><span class="op">)</span></span>
<span><span class="va">SCGEMRNA_umap</span> <span class="op">&lt;-</span> <span class="va">SCGEMRNA_umap</span><span class="op">$</span><span class="va">layout</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">SCGEMRNA_umap</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMAP1"</span>, <span class="st">"UMAP2"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">SCGEMRNA_sce</span>, <span class="st">"UMAP"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">SCGEMRNA_umap</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">SCGEMMETH_umap</span> <span class="op">&lt;-</span> <span class="fu">umap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/umap/man/umap.html" class="external-link">umap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">SCGEMMETH_sce</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"X"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span>, n_neighbors<span class="op">=</span><span class="fl">30</span>, min_dist<span class="op">=</span><span class="fl">0.7</span><span class="op">)</span></span>
<span><span class="va">SCGEMMETH_umap</span> <span class="op">&lt;-</span> <span class="va">SCGEMMETH_umap</span><span class="op">$</span><span class="va">layout</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">SCGEMMETH_umap</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMAP1"</span>, <span class="st">"UMAP2"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">SCGEMMETH_sce</span>, <span class="st">"UMAP"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">SCGEMMETH_umap</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat_RNA</span> <span class="op">&lt;-</span> <span class="va">SCGEMRNA_umap</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Method <span class="op">=</span> <span class="st">"Real data: RNA"</span><span class="op">)</span></span>
<span><span class="va">dat_METH</span> <span class="op">&lt;-</span> <span class="va">SCGEMMETH_umap</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Method <span class="op">=</span> <span class="st">"Real data: Methylation"</span><span class="op">)</span></span>
<span><span class="va">dat_NEW</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">SCGEMNEW_sce</span>, <span class="st">"UMAP"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Method <span class="op">=</span> <span class="st">"scDesign3: RNA + Meythlation"</span><span class="op">)</span></span>
<span><span class="va">SCGEM_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">dat_RNA</span>, <span class="va">dat_METH</span>, <span class="va">dat_NEW</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Then, we reformat the UMAP embeddings for the inputted scRNA-seq data, DNA methylation data, and the combined synthetic data and visualize the UMAP embeddings.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">1</span>,<span class="fl">3</span><span class="op">)</span>, <span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">dat_text_SCGEM</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>Method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Real data: RNA"</span>, <span class="st">"Real data: Methylation"</span>, <span class="st">"scDesign3: RNA + Meythlation"</span><span class="op">)</span>, label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"32 Features*177 Cells"</span>, <span class="st">"27 Features*142 Cells"</span>, <span class="st">"59 Features*319 Cells"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">SCGEM_dat</span> <span class="op">&lt;-</span> <span class="va">SCGEM_dat</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">Method</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Real data: RNA"</span>, <span class="st">"scDesign3: RNA + Meythlation"</span>, <span class="st">"Real data: Methylation"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>UMAP1 <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/if_else.html" class="external-link">if_else</a></span><span class="op">(</span><span class="va">Method</span> <span class="op">==</span> <span class="st">"Real data: RNA"</span>, <span class="op">-</span><span class="va">UMAP1</span>, <span class="va">UMAP1</span><span class="op">)</span>, UMAP2 <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/if_else.html" class="external-link">if_else</a></span><span class="op">(</span><span class="va">Method</span> <span class="op">==</span> <span class="st">"Real data: RNA"</span>, <span class="op">-</span><span class="va">UMAP2</span>, <span class="va">UMAP2</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p_merge_modals</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">SCGEM_dat</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">UMAP1</span>, <span class="va">UMAP2</span>, colour <span class="op">=</span> <span class="va">Method</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggrastr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggrastr/man/rasterize.html" class="external-link">rasterize</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>, dpi <span class="op">=</span> <span class="fl">300</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_color_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set2"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>aspect.ratio <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    panel.grid.minor <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    panel.grid.major <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.x<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.ticks.x<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.y<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.ticks.y<span class="op">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://teunbrand.github.io/ggh4x/reference/facet_manual.html" class="external-link">facet_manual</a></span><span class="op">(</span><span class="op">~</span><span class="va">Method</span>, design <span class="op">=</span> <span class="va">design</span>, widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>, heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, respect <span class="op">=</span> <span class="cn">TRUE</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html" class="external-link">geom_text</a></span><span class="op">(</span></span>
<span>  data    <span class="op">=</span> <span class="va">dat_text_SCGEM</span>,</span>
<span>  mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="cn">Inf</span>, y <span class="op">=</span> <span class="op">-</span><span class="cn">Inf</span>, label <span class="op">=</span> <span class="va">label</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">6</span>, hjust <span class="op">=</span> <span class="fl">1</span>, color <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">p_merge_modals</span></span></code></pre></div>
<p><img src="scDesign3-multiomics-vignette_files/figure-html/unnamed-chunk-16-1.png" width="480"></p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 20.04.6 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/Los_Angeles</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] umap_0.2.10.0               ggh4x_0.2.4                </span></span>
<span><span class="co">#&gt;  [3] ggplot2_3.4.2               dplyr_1.1.2                </span></span>
<span><span class="co">#&gt;  [5] SingleCellExperiment_1.22.0 SummarizedExperiment_1.30.2</span></span>
<span><span class="co">#&gt;  [7] Biobase_2.60.0              GenomicRanges_1.52.0       </span></span>
<span><span class="co">#&gt;  [9] GenomeInfoDb_1.36.1         IRanges_2.34.1             </span></span>
<span><span class="co">#&gt; [11] S4Vectors_0.38.1            BiocGenerics_0.46.0        </span></span>
<span><span class="co">#&gt; [13] MatrixGenerics_1.12.2       matrixStats_1.0.0          </span></span>
<span><span class="co">#&gt; [15] scDesign3_0.99.5            BiocStyle_2.28.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.0        farver_2.1.1            vipor_0.4.5            </span></span>
<span><span class="co">#&gt;  [4] bitops_1.0-7            kde1d_1.0.5             fastmap_1.1.1          </span></span>
<span><span class="co">#&gt;  [7] RCurl_1.98-1.12         digest_0.6.33           lifecycle_1.0.3        </span></span>
<span><span class="co">#&gt; [10] Cairo_1.6-0             survival_3.5-5          gamlss.dist_6.0-5      </span></span>
<span><span class="co">#&gt; [13] magrittr_2.0.3          compiler_4.3.0          rlang_1.1.1            </span></span>
<span><span class="co">#&gt; [16] sass_0.4.6              tools_4.3.0             utf8_1.2.3             </span></span>
<span><span class="co">#&gt; [19] yaml_2.3.7              knitr_1.43              labeling_0.4.2         </span></span>
<span><span class="co">#&gt; [22] askpass_1.1             S4Arrays_1.0.4          mclust_6.0.0           </span></span>
<span><span class="co">#&gt; [25] reticulate_1.30         DelayedArray_0.26.6     RColorBrewer_1.1-3     </span></span>
<span><span class="co">#&gt; [28] withr_2.5.0             purrr_1.0.1             rvinecopulib_0.6.3.1.1 </span></span>
<span><span class="co">#&gt; [31] desc_1.4.2              grid_4.3.0              fansi_1.0.4            </span></span>
<span><span class="co">#&gt; [34] colorspace_2.1-0        scales_1.2.1            MASS_7.3-60            </span></span>
<span><span class="co">#&gt; [37] cli_3.6.1               rmarkdown_2.23          crayon_1.5.2           </span></span>
<span><span class="co">#&gt; [40] ragg_1.2.5              generics_0.1.3          RSpectra_0.16-1        </span></span>
<span><span class="co">#&gt; [43] ggbeeswarm_0.7.2        cachem_1.0.8            stringr_1.5.0          </span></span>
<span><span class="co">#&gt; [46] zlibbioc_1.46.0         splines_4.3.0           assertthat_0.2.1       </span></span>
<span><span class="co">#&gt; [49] parallel_4.3.0          ggrastr_1.0.2           BiocManager_1.30.21    </span></span>
<span><span class="co">#&gt; [52] XVector_0.40.0          vctrs_0.6.3             Matrix_1.6-0           </span></span>
<span><span class="co">#&gt; [55] jsonlite_1.8.7          bookdown_0.34           gamlss_5.4-12          </span></span>
<span><span class="co">#&gt; [58] beeswarm_0.4.0          irlba_2.3.5.1           systemfonts_1.0.4      </span></span>
<span><span class="co">#&gt; [61] jquerylib_0.1.4         glue_1.6.2              pkgdown_2.0.7          </span></span>
<span><span class="co">#&gt; [64] rngWELL_0.10-9          stringi_1.7.12          gtable_0.3.3           </span></span>
<span><span class="co">#&gt; [67] randtoolbox_2.0.4       munsell_0.5.0           tibble_3.2.1           </span></span>
<span><span class="co">#&gt; [70] pillar_1.9.0            htmltools_0.5.5         openssl_2.0.6          </span></span>
<span><span class="co">#&gt; [73] gamlss.data_6.0-2       GenomeInfoDbData_1.2.10 R6_2.5.1               </span></span>
<span><span class="co">#&gt; [76] textshaping_0.3.6       rprojroot_2.0.3         evaluate_0.21          </span></span>
<span><span class="co">#&gt; [79] lattice_0.21-8          highr_0.10              png_0.1-8              </span></span>
<span><span class="co">#&gt; [82] memoise_2.0.1           bslib_0.5.0             Rcpp_1.0.11            </span></span>
<span><span class="co">#&gt; [85] nlme_3.1-162            mgcv_1.8-42             xfun_0.39              </span></span>
<span><span class="co">#&gt; [88] fs_1.6.2                pkgconfig_2.0.3</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Dongyuan Song, Qingyang Wang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
