<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scDesign3">
<title>scDesign3 Quickstart • scDesign3</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="scDesign3 Quickstart">
<meta property="og:description" content="scDesign3">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scDesign3</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/scDesign3.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/scDesign3-introduction-vignette.html">scDesign3 Introduction</a>
    <a class="dropdown-item" href="../articles/scDesign3-marginal-vignette.html">scDesign3 marginal distribution for genes</a>
    <a class="dropdown-item" href="../articles/scDesign3-copulaCompare-vignette.html">Compare gaussian copula and vine copula</a>
    <a class="dropdown-item" href="../articles/scDesign3-parallelization-vignette.html">Parallelization in scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-librarySize-vignette.html">Simulate datasets with cell library size</a>
    <a class="dropdown-item" href="../articles/scDesign3-multipleLineages-vignette.html">Simulate datasets with multiple lineages</a>
    <a class="dropdown-item" href="../articles/scDesign3-conditionEffect-vignette.html">Simulate datasets with condition effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-batchEffect-vignette.html">Simulate datasets with batch effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-spatial-vignette.html">Simulate spatial transcriptomic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-spatial-deconvolution.html">Simulate spot-resolution spatial data for cell-type deconvolution</a>
    <a class="dropdown-item" href="../articles/scDesign3-scATACseq-vignette.html">Simulate single-cell ATAC-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-CITEseq-vignette.html">Simulate CITE-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-multiomics-vignette.html">Simulate multi-omics data from single-omic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-clusterGOF-vignette.html">Evaluate clustering goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-pseudotimeGOF-vignette.html">Evaluate pseudotime goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-DEanalysis-vignette.html">Benchmarking DE Analysis with scDesign3</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/SONGDONGYUAN1994/scDesign3/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>scDesign3 Quickstart</h1>
                        <h4 data-toc-skip class="author">Dongyuan
Song</h4>
            <address class="author_afil">
      Bioinformatics IDP, University of California, Los
Angeles<br><a class="author_email" href="mailto:#"></a><a href="mailto:dongyuansong@ucla.edu" class="email">dongyuansong@ucla.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Qingyang
Wang</h4>
            <address class="author_afil">
      Department of Statistics, University of California, Los
Angeles<br><a class="author_email" href="mailto:#"></a><a href="mailto:qw802@g.ucla.edu" class="email">qw802@g.ucla.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">21 August 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/SONGDONGYUAN1994/scDesign3/blob/HEAD/../../vignettes/scDesign3.Rmd" class="external-link"><code>../../vignettes/scDesign3.Rmd</code></a></small>
      <div class="d-none name"><code>scDesign3.Rmd</code></div>
    </div>

    
    
<style type="text/css">
pre {
  white-space: pre !important;
  overflow-x: scroll !important;
}
</style>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/SONGDONGYUAN1994/scDesign3" class="external-link">scDesign3</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>scDesign3 is a unified probabilistic framework that generates
realistic in silico high-dimensional single-cell omics data of various
cell states, including discrete cell types, continuous trajectories, and
spatial locations by learning from real datasets. Since the functions of
scDesign3 is very comprehensive, here we only introduce how scDesign3
simulates an scRNA-seq dataset with one continuous developmental
trajectory. For more information, please check the Articles on our
website: (<a href="https://songdongyuan1994.github.io/scDesign3/docs/index.html" class="external-link uri">https://songdongyuan1994.github.io/scDesign3/docs/index.html</a>).</p>
</div>
<div class="section level2">
<h2 id="read-in-the-reference-data">Read in the reference data<a class="anchor" aria-label="anchor" href="#read-in-the-reference-data"></a>
</h2>
<p>The raw data is from the <a href="https://scvelo.readthedocs.io/scvelo.datasets.pancreas/" class="external-link">scvelo</a>,
which describes pancreatic endocrinogenesis. We pre-select the top 1000
highly variable genes and filter out some cell types to ensure a
<strong>single trajectory</strong>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://figshare.com/ndownloader/files/40581992"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 1000 2087 </span></span>
<span><span class="co">#&gt; metadata(5): clusters_coarse_colors clusters_colors day_colors</span></span>
<span><span class="co">#&gt;   neighbors pca</span></span>
<span><span class="co">#&gt; assays(6): X spliced ... cpm logcounts</span></span>
<span><span class="co">#&gt; rownames(1000): Pyy Iapp ... Eya2 Kif21a</span></span>
<span><span class="co">#&gt; rowData names(1): highly_variable_genes</span></span>
<span><span class="co">#&gt; colnames(2087): AAACCTGAGAGGGATA AAACCTGGTAAGTGGC ... TTTGTCAAGTGACATA</span></span>
<span><span class="co">#&gt;   TTTGTCAAGTGTGGCA</span></span>
<span><span class="co">#&gt; colData names(7): clusters_coarse clusters ... sizeFactor pseudotime</span></span>
<span><span class="co">#&gt; reducedDimNames(4): X_pca X_umap PCA UMAP</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code></pre></div>
<p>To save computational time, we only use the top 100 genes.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_sce</span> <span class="op">&lt;-</span> <span class="va">example_sce</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span>, <span class="op">]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="simulation">Simulation<a class="anchor" aria-label="anchor" href="#simulation"></a>
</h2>
<p>The function <code><a href="../reference/scdesign3.html">scdesign3()</a></code> takes in a
<code>SinglecellExperiment</code> object with the cell covariates (such
as cell types, pseudotime, or spatial coordinates) stored in the
<code>colData</code> of the <code>SinglecellExperiment</code>
object.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">example_simu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scdesign3.html">scdesign3</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">example_sce</span>,</span>
<span>    assay_use <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>    celltype <span class="op">=</span> <span class="st">"cell_type"</span>,</span>
<span>    pseudotime <span class="op">=</span> <span class="st">"pseudotime"</span>,</span>
<span>    spatial <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    other_covariates <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    mu_formula <span class="op">=</span> <span class="st">"s(pseudotime, k = 10, bs = 'cr')"</span>,</span>
<span>    sigma_formula <span class="op">=</span> <span class="st">"1"</span>, <span class="co"># If you want your dispersion also varies along pseudotime, use "s(pseudotime, k = 5, bs = 'cr')"</span></span>
<span>    family_use <span class="op">=</span> <span class="st">"nb"</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    usebam <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    corr_formula <span class="op">=</span> <span class="st">"1"</span>,</span>
<span>    copula <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>    DT <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    pseudo_obs <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    return_model <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    nonzerovar <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>The output of <code><a href="../reference/scdesign3.html">scdesign3()</a></code> is a list which includes:</p>
<ul>
<li>
<code>new_count</code>: This is the synthetic count matrix generated
by <code><a href="../reference/scdesign3.html">scdesign3()</a></code>.</li>
<li>
<code>new_covariate</code>:
<ul>
<li>If the parameter <code>ncell</code> is set to a number that is
different from the number of cells in the input data, this will be a
matrix that has the new cell covariates that are used for generating new
data.</li>
<li>If the parameter <code>ncell</code> is the default value, this will
be <code>NULL</code>.</li>
</ul>
</li>
<li>
<code>model_aic</code>: This is a vector include the genes’ marginal
models’ AIC, fitted copula’s AIC, and total AIC, which is the sum of the
previous two AIC.</li>
<li>
<code>model_bic</code>: This is a vector include the genes’ marginal
models’ BIC, fitted copula’s BIC, and total BIC, which is the sum of the
previous two BIC.</li>
<li>
<code>marginal_list</code>:
<ul>
<li>If the parameter <code>return_model</code> is set to
<code>TRUE</code>, this will be a list which contains the fitted gam or
gamlss model for all genes in the input data.</li>
<li>If the parameter <code>return_model</code> is set to the default
value <code>FALSE</code>, this will be <code>NULL</code>.</li>
</ul>
</li>
<li>
<code>corr_list</code>:
<ul>
<li>If the parameter <code>return_model</code> is set to
<code>TRUE</code>, this will be a list which contains the either a
correlation matrix (when <code>copula = "gaussian"</code>) or the fitted
vine copula (when <code>copula = "vine</code>) for each user specified
correlation groups (based on the parameter <code>corr_by</code>).</li>
<li>If the parameter <code>return_model</code> is set to the default
value <code>FALSE</code>, this will be <code>NULL</code>.</li>
</ul>
</li>
</ul>
<p>In this example, since we did not change the parameter
<code>ncell</code>, the synthetic count matrix will have the same
dimension as the input count matrix.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">example_simu</span><span class="op">$</span><span class="va">new_count</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1]  100 2087</span></span></code></pre></div>
<p>Then, we can create the <code>SinglecellExperiment</code> object
using the synthetic count matrix and store the <code>logcounts</code> to
the input and synthetic <code>SinglecellExperiment</code> objects.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html" class="external-link">logcounts</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">simu_sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">example_simu</span><span class="op">$</span><span class="va">new_count</span><span class="op">)</span>, colData <span class="op">=</span> <span class="va">example_simu</span><span class="op">$</span><span class="va">new_covariate</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html" class="external-link">logcounts</a></span><span class="op">(</span><span class="va">simu_sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">simu_sce</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">compare_figure</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_reduceddim.html">plot_reduceddim</a></span><span class="op">(</span>ref_sce <span class="op">=</span> <span class="va">example_sce</span>, </span>
<span>                                  sce_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">simu_sce</span><span class="op">)</span>, </span>
<span>                                  name_vec <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Reference"</span>, <span class="st">"scDesign3"</span><span class="op">)</span>,</span>
<span>                                  assay_use <span class="op">=</span> <span class="st">"logcounts"</span>, </span>
<span>                                  if_plot <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                  color_by <span class="op">=</span> <span class="st">"pseudotime"</span>, </span>
<span>                                  n_pc <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">compare_figure</span><span class="op">$</span><span class="va">p_umap</span><span class="op">)</span></span></code></pre></div>
<p><img src="scDesign3_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 20.04.6 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/Los_Angeles</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] ggplot2_3.5.0               SingleCellExperiment_1.22.0</span></span>
<span><span class="co">#&gt;  [3] SummarizedExperiment_1.30.2 Biobase_2.60.0             </span></span>
<span><span class="co">#&gt;  [5] GenomicRanges_1.52.0        GenomeInfoDb_1.36.1        </span></span>
<span><span class="co">#&gt;  [7] IRanges_2.34.1              S4Vectors_0.38.1           </span></span>
<span><span class="co">#&gt;  [9] BiocGenerics_0.46.0         MatrixGenerics_1.12.2      </span></span>
<span><span class="co">#&gt; [11] matrixStats_1.0.0           scDesign3_1.1.3            </span></span>
<span><span class="co">#&gt; [13] BiocStyle_2.28.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.0        viridisLite_0.4.2       farver_2.1.1           </span></span>
<span><span class="co">#&gt;  [4] dplyr_1.1.4             viridis_0.6.4           bitops_1.0-7           </span></span>
<span><span class="co">#&gt;  [7] fastmap_1.1.1           RCurl_1.98-1.12         coop_0.6-3             </span></span>
<span><span class="co">#&gt; [10] digest_0.6.33           lifecycle_1.0.4         survival_3.5-8         </span></span>
<span><span class="co">#&gt; [13] gamlss.dist_6.0-5       magrittr_2.0.3          compiler_4.4.0         </span></span>
<span><span class="co">#&gt; [16] rlang_1.1.3             sass_0.4.7              tools_4.4.0            </span></span>
<span><span class="co">#&gt; [19] utf8_1.2.4              yaml_2.3.7              knitr_1.43             </span></span>
<span><span class="co">#&gt; [22] labeling_0.4.3          askpass_1.1             S4Arrays_1.2.0         </span></span>
<span><span class="co">#&gt; [25] mclust_6.0.1            reticulate_1.30         DelayedArray_0.26.6    </span></span>
<span><span class="co">#&gt; [28] abind_1.4-5             withr_3.0.0             purrr_1.0.1            </span></span>
<span><span class="co">#&gt; [31] desc_1.4.2              grid_4.4.0              fansi_1.0.6            </span></span>
<span><span class="co">#&gt; [34] colorspace_2.1-0        scales_1.3.0            MASS_7.3-60.0.1        </span></span>
<span><span class="co">#&gt; [37] cli_3.6.2               mvtnorm_1.2-2           rmarkdown_2.23         </span></span>
<span><span class="co">#&gt; [40] crayon_1.5.2            ragg_1.2.5              generics_0.1.3         </span></span>
<span><span class="co">#&gt; [43] umap_0.2.10.0           rstudioapi_0.15.0       RSpectra_0.16-2        </span></span>
<span><span class="co">#&gt; [46] cachem_1.0.8            stringr_1.5.1           zlibbioc_1.46.0        </span></span>
<span><span class="co">#&gt; [49] splines_4.4.0           parallel_4.4.0          BiocManager_1.30.21.1  </span></span>
<span><span class="co">#&gt; [52] XVector_0.40.0          vctrs_0.6.5             Matrix_1.7-0           </span></span>
<span><span class="co">#&gt; [55] jsonlite_1.8.7          bookdown_0.34           gamlss_5.4-12          </span></span>
<span><span class="co">#&gt; [58] irlba_2.3.5.1           systemfonts_1.0.4       jquerylib_0.1.4        </span></span>
<span><span class="co">#&gt; [61] glue_1.7.0              pkgdown_2.0.7           stringi_1.8.3          </span></span>
<span><span class="co">#&gt; [64] gtable_0.3.4            munsell_0.5.0           tibble_3.2.1           </span></span>
<span><span class="co">#&gt; [67] pillar_1.9.0            htmltools_0.5.5         openssl_2.1.0          </span></span>
<span><span class="co">#&gt; [70] gamlss.data_6.0-2       GenomeInfoDbData_1.2.10 R6_2.5.1               </span></span>
<span><span class="co">#&gt; [73] textshaping_0.3.6       rprojroot_2.0.3         evaluate_0.21          </span></span>
<span><span class="co">#&gt; [76] lattice_0.22-5          highr_0.10              png_0.1-8              </span></span>
<span><span class="co">#&gt; [79] memoise_2.0.1           bslib_0.5.0             Rcpp_1.0.12            </span></span>
<span><span class="co">#&gt; [82] gridExtra_2.3           nlme_3.1-163            mgcv_1.9-1             </span></span>
<span><span class="co">#&gt; [85] xfun_0.39               fs_1.6.3                pkgconfig_2.0.3</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Dongyuan Song, Qingyang Wang, Chenxin Jiang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
