<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scDesign3">
<title>Simulate datasets with batch effect • scDesign3</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Simulate datasets with batch effect">
<meta property="og:description" content="scDesign3">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scDesign3</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.5</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/scDesign3.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/scDesign3-introduction-vignette.html">scDesign3 Introduction</a>
    <a class="dropdown-item" href="../articles/scDesign3-marginal-vignette.html">scDesign3 marginal distribution for genes</a>
    <a class="dropdown-item" href="../articles/scDesign3-copulaCompare-vignette.html">Compare gaussian copula and vine copula</a>
    <a class="dropdown-item" href="../articles/scDesign3-librarySize-vignette.html">Simulate datasets with cell library size</a>
    <a class="dropdown-item" href="../articles/scDesign3-multipleLineages-vignette.html">Simulate datasets with multiple lineages</a>
    <a class="dropdown-item" href="../articles/scDesign3-conditionEffect-vignette.html">Simulate datasets with condition effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-batchEffect-vignette.html">Simulate datasets with batch effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-spatial-vignette.html">Simulate spatial transcriptomic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-spatial-deconvolution.html">Simulate spot-resolution spatial data for cell-type deconvolution</a>
    <a class="dropdown-item" href="../articles/scDesign3-scATACseq-vignette.html">Simulate single-cell ATAC-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-CITEseq-vignette.html">Simulate CITE-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-multiomics-vignette.html">Simulate multi-omics data from single-omic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-clusterGOF-vignette.html">Evaluate clustering goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-pseudotimeGOF-vignette.html">Evaluate pseudotime goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-DEanalysis-vignette.html">Benchmarking DE Analysis with scDesign3</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/SONGDONGYUAN1994/scDesign3/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">

<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Simulate datasets with batch effect</h1>
                        <h4 data-toc-skip class="author">Dongyuan Song</h4>
            <address class="author_afil">
      Bioinformatics IDP, University of California, Los Angeles<br><a class="author_email" href="mailto:#"></a><a href="mailto:dongyuansong@ucla.edu" class="email">dongyuansong@ucla.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Qingyang Wang</h4>
            <address class="author_afil">
      Department of Statistics, University of California, Los Angeles<br><a class="author_email" href="mailto:#"></a><a href="mailto:qw802@g.ucla.edu" class="email">qw802@g.ucla.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">15 July 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/SONGDONGYUAN1994/scDesign3/blob/HEAD/../../scDesign3/code/vignettes/scDesign3-batchEffect-vignette.Rmd" class="external-link"><code>../../scDesign3/code/vignettes/scDesign3-batchEffect-vignette.Rmd</code></a></small>
      <div class="d-none name"><code>scDesign3-batchEffect-vignette.Rmd</code></div>
    </div>

    
    
<style type="text/css">
pre {
  white-space: pre !important;
  overflow-x: scroll !important;
}
</style>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/SONGDONGYUAN1994/scDesign3" class="external-link">scDesign3</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In this tutorial, we will show how to use scDesign3 to simulate data with original batch effects and how to remove the batch effects. We will also demostrate how to add ariticial batch effects.</p>
</div>
<div class="section level2">
<h2 id="read-in-the-reference-data">Read in the reference data<a class="anchor" aria-label="anchor" href="#read-in-the-reference-data"></a>
</h2>
<p>The raw data is from the <code>SeuratData</code> package. The data is called <code>pbmcsca</code> in the package; it is PBMC Systematic Comparative Analysis dataset from the Broad Institute.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://figshare.com/ndownloader/files/40581965"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 1000 6276 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span><span class="co">#&gt; rownames(1000): LYZ GNLY ... PBXIP1 SECTM1</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(6276): pbmc2_10X_V2_AAACCTGAGATGGGTC</span></span>
<span><span class="co">#&gt;   pbmc2_10X_V2_AAACCTGAGCGTAATA ... pbmc1_10x_v3_TTGAACGCATGCAGCC</span></span>
<span><span class="co">#&gt;   pbmc1_10x_v3_TTTGACTAGTGTTCCA</span></span>
<span><span class="co">#&gt; colData names(13): phenoid orig.ident ... batch cell_type</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span></code></pre></div>
<p>To save computational time, we only use the top 100 genes.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_sce</span> <span class="op">&lt;-</span> <span class="va">example_sce</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span>, <span class="op">]</span></span></code></pre></div>
<p>The column <code>batch</code> in this example dataset’s colData contains the batch information.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 6 rows and 13 columns</span></span>
<span><span class="co">#&gt;                                        phenoid orig.ident nCount_RNA</span></span>
<span><span class="co">#&gt;                                    &lt;character&gt;   &lt;factor&gt;  &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGATGGGTC           B cell      pbmc2       2360</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGCGTAATA           B cell      pbmc2       1888</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGCTAGGCA Cytotoxic T cell      pbmc2       3456</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGGGTCTCC   Dendritic cell      pbmc2       3802</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGGTCCGAACC      CD4+ T cell      pbmc2       3826</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGTCGTCCGTT      CD4+ T cell      pbmc2       2345</span></span>
<span><span class="co">#&gt;                               nFeature_RNA       nGene        nUMI</span></span>
<span><span class="co">#&gt;                                  &lt;integer&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGATGGGTC         1044        1044        2360</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGCGTAATA          803         803        1888</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGCTAGGCA         1372        1372        3456</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGGGTCTCC         1519        1519        3802</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGGTCCGAACC         1451        1451        3826</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGTCGTCCGTT          931         931        2345</span></span>
<span><span class="co">#&gt;                                     percent.mito     Cluster  Experiment</span></span>
<span><span class="co">#&gt;                                      &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGATGGGTC 0.0419491525423729           2       pbmc2</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGCGTAATA 0.0413135593220339           2       pbmc2</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGCTAGGCA 0.0353009259259259           1       pbmc2</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGGGTCTCC 0.0420831141504471           6       pbmc2</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGGTCCGAACC 0.0371144798745426           0       pbmc2</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGTCGTCCGTT 0.0652452025586354           0       pbmc2</span></span>
<span><span class="co">#&gt;                                          Method    ident             batch</span></span>
<span><span class="co">#&gt;                                     &lt;character&gt; &lt;factor&gt;       &lt;character&gt;</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGATGGGTC 10x Chromium (v2)    pbmc2 10x Chromium (v2)</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGCGTAATA 10x Chromium (v2)    pbmc2 10x Chromium (v2)</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGCTAGGCA 10x Chromium (v2)    pbmc2 10x Chromium (v2)</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGGGTCTCC 10x Chromium (v2)    pbmc2 10x Chromium (v2)</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGGTCCGAACC 10x Chromium (v2)    pbmc2 10x Chromium (v2)</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGTCGTCCGTT 10x Chromium (v2)    pbmc2 10x Chromium (v2)</span></span>
<span><span class="co">#&gt;                                      cell_type</span></span>
<span><span class="co">#&gt;                                    &lt;character&gt;</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGATGGGTC           B cell</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGCGTAATA           B cell</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGCTAGGCA Cytotoxic T cell</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGAGGGTCTCC   Dendritic cell</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGGTCCGAACC      CD4+ T cell</span></span>
<span><span class="co">#&gt; pbmc2_10X_V2_AAACCTGTCGTCCGTT      CD4+ T cell</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="simulation">Simulation<a class="anchor" aria-label="anchor" href="#simulation"></a>
</h2>
<p>We can simulate a new data with batch effect information.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">simu_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scdesign3.html">scdesign3</a></span><span class="op">(</span>sce <span class="op">=</span> <span class="va">example_sce</span>, </span>
<span>                              assay_use <span class="op">=</span> <span class="st">"counts"</span>, </span>
<span>                              celltype <span class="op">=</span> <span class="st">"cell_type"</span>, </span>
<span>                              pseudotime <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                              spatial <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                              other_covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"batch"</span><span class="op">)</span>, </span>
<span>                              mu_formula <span class="op">=</span> <span class="st">"cell_type + batch"</span>, </span>
<span>                              sigma_formula <span class="op">=</span> <span class="st">"1"</span>, </span>
<span>                              family_use <span class="op">=</span> <span class="st">"nb"</span>, </span>
<span>                              n_cores <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                              usebam <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                              corr_formula <span class="op">=</span> <span class="st">"1"</span>, </span>
<span>                              copula <span class="op">=</span> <span class="st">"gaussian"</span>, </span>
<span>                              DT <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                              pseudo_obs <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>                              return_model <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>We can also remove the batch effect and generate new data.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">BATCH_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/construct_data.html">construct_data</a></span><span class="op">(</span></span>
<span>  sce <span class="op">=</span> <span class="va">example_sce</span>,</span>
<span>  assay_use <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  celltype <span class="op">=</span> <span class="st">"cell_type"</span>,</span>
<span>  pseudotime <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  spatial <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  other_covariates <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"batch"</span><span class="op">)</span>,</span>
<span>  corr_by <span class="op">=</span> <span class="st">"1"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">BATCH_marginal</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fit_marginal.html">fit_marginal</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">BATCH_data</span>,</span>
<span>  predictor <span class="op">=</span> <span class="st">"gene"</span>,</span>
<span>  mu_formula <span class="op">=</span> <span class="st">"cell_type + batch"</span>,</span>
<span>  sigma_formula <span class="op">=</span> <span class="st">"1"</span>,</span>
<span>  family_use <span class="op">=</span> <span class="st">"nb"</span>,</span>
<span>  n_cores <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  usebam <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">BATCH_copula</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fit_copula.html">fit_copula</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">example_sce</span>,</span>
<span>    assay_use <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>    marginal_list <span class="op">=</span> <span class="va">BATCH_marginal</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"nb"</span>,</span>
<span>    copula <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    input_data <span class="op">=</span> <span class="va">BATCH_data</span><span class="op">$</span><span class="va">dat</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>In here, we remove the batch effect by setting its coefficient to zero for all genes’ marginal fits. Then, we use the new sets of coefficients to generate the parameters for all genes across all cells.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">BATCH_marginal_null</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BATCH_marginal</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">x</span><span class="op">$</span><span class="va">fit</span><span class="op">$</span><span class="va">coefficients</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">fit</span><span class="op">$</span><span class="va">coefficients</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span>  <span class="va">x</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">BATCH_para_null</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_para.html">extract_para</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">example_sce</span>,</span>
<span>    marginal_list <span class="op">=</span> <span class="va">BATCH_marginal_null</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"nb"</span>,</span>
<span>    new_covariate <span class="op">=</span>  <span class="va">BATCH_data</span><span class="op">$</span><span class="va">newCovariate</span>,</span>
<span>    data <span class="op">=</span> <span class="va">BATCH_data</span><span class="op">$</span><span class="va">dat</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">BATCH_newcount_null</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simu_new.html">simu_new</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">example_sce</span>,</span>
<span>    mean_mat <span class="op">=</span> <span class="va">BATCH_para_null</span><span class="op">$</span><span class="va">mean_mat</span>,</span>
<span>    sigma_mat <span class="op">=</span> <span class="va">BATCH_para_null</span><span class="op">$</span><span class="va">sigma_mat</span>,</span>
<span>    zero_mat <span class="op">=</span> <span class="va">BATCH_para_null</span><span class="op">$</span><span class="va">zero_mat</span>,</span>
<span>    quantile_mat <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    copula_list <span class="op">=</span> <span class="va">BATCH_copula</span><span class="op">$</span><span class="va">copula_list</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"nb"</span>,</span>
<span>    input_data <span class="op">=</span> <span class="va">BATCH_data</span><span class="op">$</span><span class="va">dat</span>,</span>
<span>    new_covariate <span class="op">=</span> <span class="va">BATCH_data</span><span class="op">$</span><span class="va">newCovariate</span>,</span>
<span>    important_feature <span class="op">=</span> <span class="va">BATCH_copula</span><span class="op">$</span><span class="va">important_feature</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>Additionally, we can alter the batch effect information by mannually change the estimated coefficient for batch effect in each gene’s marginal model. Then, we can simulate new dataset with altered batch effect information.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">BATCH_marginal_alter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">BATCH_marginal</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">x</span><span class="op">$</span><span class="va">fit</span><span class="op">$</span><span class="va">coefficients</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">fit</span><span class="op">$</span><span class="va">coefficients</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">1</span>, mean <span class="op">=</span> <span class="fl">5</span>, sd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span>  <span class="va">x</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">BATCH_para_alter</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_para.html">extract_para</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">example_sce</span>,</span>
<span>    marginal_list <span class="op">=</span> <span class="va">BATCH_marginal_alter</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"nb"</span>,</span>
<span>    new_covariate <span class="op">=</span> <span class="va">BATCH_data</span><span class="op">$</span><span class="va">newCovariate</span>,</span>
<span>    data <span class="op">=</span> <span class="va">BATCH_data</span><span class="op">$</span><span class="va">dat</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">BATCH_newcount_alter</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simu_new.html">simu_new</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">example_sce</span>,</span>
<span>    mean_mat <span class="op">=</span> <span class="va">BATCH_para_alter</span><span class="op">$</span><span class="va">mean_mat</span>,</span>
<span>    sigma_mat <span class="op">=</span> <span class="va">BATCH_para_alter</span><span class="op">$</span><span class="va">sigma_mat</span>,</span>
<span>    zero_mat <span class="op">=</span> <span class="va">BATCH_para_alter</span><span class="op">$</span><span class="va">zero_mat</span>,</span>
<span>    quantile_mat <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    copula_list <span class="op">=</span> <span class="va">BATCH_copula</span><span class="op">$</span><span class="va">copula_list</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"nb"</span>,</span>
<span>    input_data <span class="op">=</span> <span class="va">BATCH_data</span><span class="op">$</span><span class="va">dat</span>,</span>
<span>    new_covariate <span class="op">=</span> <span class="va">BATCH_data</span><span class="op">$</span><span class="va">newCovariate</span>,</span>
<span>    important_feature <span class="op">=</span> <span class="va">BATCH_copula</span><span class="op">$</span><span class="va">important_feature</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>We create three <code>SingleCellExperiment</code> objects using three count matrices generated above and store their <code>logcounts</code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">simu_res_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">simu_res</span><span class="op">$</span><span class="va">new_count</span>, <span class="va">BATCH_newcount_null</span>, <span class="va">BATCH_newcount_alter</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">simu_sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">x</span><span class="op">)</span>, colData <span class="op">=</span> <span class="va">BATCH_data</span><span class="op">$</span><span class="va">newCovariate</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html" class="external-link">logcounts</a></span><span class="op">(</span><span class="va">simu_sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">simu_sce</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">simu_sce</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visulization">Visulization<a class="anchor" aria-label="anchor" href="#visulization"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">compare_figure</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_reduceddim.html">plot_reduceddim</a></span><span class="op">(</span>ref_sce <span class="op">=</span> <span class="va">example_sce</span>, </span>
<span>                                  sce_list <span class="op">=</span> <span class="va">simu_res_list</span>, </span>
<span>                                  name_vec <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Reference"</span>, <span class="st">"w/ Batch"</span>, <span class="st">"w/o Batch"</span>,<span class="st">"Aritifical Batch"</span><span class="op">)</span>,</span>
<span>                                  assay_use <span class="op">=</span> <span class="st">"logcounts"</span>, </span>
<span>                                  if_plot <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                  color_by <span class="op">=</span> <span class="st">"cell_type"</span>, </span>
<span>                                  shape_by <span class="op">=</span> <span class="st">"batch"</span>,</span>
<span>                                  n_pc <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">compare_figure</span><span class="op">$</span><span class="va">p_umap</span><span class="op">)</span></span></code></pre></div>
<p><img src="scDesign3-batchEffect-vignette_files/figure-html/unnamed-chunk-17-1.png" width="1152"></p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 20.04.6 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/Los_Angeles</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] SingleCellExperiment_1.22.0 SummarizedExperiment_1.30.2</span></span>
<span><span class="co">#&gt;  [3] Biobase_2.60.0              GenomicRanges_1.52.0       </span></span>
<span><span class="co">#&gt;  [5] GenomeInfoDb_1.36.1         IRanges_2.34.1             </span></span>
<span><span class="co">#&gt;  [7] S4Vectors_0.38.1            BiocGenerics_0.46.0        </span></span>
<span><span class="co">#&gt;  [9] MatrixGenerics_1.12.2       matrixStats_1.0.0          </span></span>
<span><span class="co">#&gt; [11] ggplot2_3.4.2               scDesign3_0.99.5           </span></span>
<span><span class="co">#&gt; [13] BiocStyle_2.28.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.0        farver_2.1.1            dplyr_1.1.2            </span></span>
<span><span class="co">#&gt;  [4] bitops_1.0-7            fastmap_1.1.1           RCurl_1.98-1.12        </span></span>
<span><span class="co">#&gt;  [7] digest_0.6.33           lifecycle_1.0.3         survival_3.5-5         </span></span>
<span><span class="co">#&gt; [10] gamlss.dist_6.0-5       magrittr_2.0.3          compiler_4.3.0         </span></span>
<span><span class="co">#&gt; [13] rlang_1.1.1             sass_0.4.6              tools_4.3.0            </span></span>
<span><span class="co">#&gt; [16] utf8_1.2.3              yaml_2.3.7              knitr_1.43             </span></span>
<span><span class="co">#&gt; [19] labeling_0.4.2          askpass_1.1             S4Arrays_1.0.4         </span></span>
<span><span class="co">#&gt; [22] mclust_6.0.0            reticulate_1.30         DelayedArray_0.26.6    </span></span>
<span><span class="co">#&gt; [25] withr_2.5.0             purrr_1.0.1             desc_1.4.2             </span></span>
<span><span class="co">#&gt; [28] grid_4.3.0              fansi_1.0.4             colorspace_2.1-0       </span></span>
<span><span class="co">#&gt; [31] scales_1.2.1            MASS_7.3-60             cli_3.6.1              </span></span>
<span><span class="co">#&gt; [34] mvtnorm_1.2-2           rmarkdown_2.23          crayon_1.5.2           </span></span>
<span><span class="co">#&gt; [37] ragg_1.2.5              generics_0.1.3          umap_0.2.10.0          </span></span>
<span><span class="co">#&gt; [40] RSpectra_0.16-1         cachem_1.0.8            stringr_1.5.0          </span></span>
<span><span class="co">#&gt; [43] zlibbioc_1.46.0         splines_4.3.0           parallel_4.3.0         </span></span>
<span><span class="co">#&gt; [46] BiocManager_1.30.21     XVector_0.40.0          vctrs_0.6.3            </span></span>
<span><span class="co">#&gt; [49] Matrix_1.6-0            jsonlite_1.8.7          bookdown_0.34          </span></span>
<span><span class="co">#&gt; [52] gamlss_5.4-12           irlba_2.3.5.1           systemfonts_1.0.4      </span></span>
<span><span class="co">#&gt; [55] jquerylib_0.1.4         glue_1.6.2              pkgdown_2.0.7          </span></span>
<span><span class="co">#&gt; [58] stringi_1.7.12          gtable_0.3.3            munsell_0.5.0          </span></span>
<span><span class="co">#&gt; [61] tibble_3.2.1            pillar_1.9.0            htmltools_0.5.5        </span></span>
<span><span class="co">#&gt; [64] openssl_2.0.6           gamlss.data_6.0-2       GenomeInfoDbData_1.2.10</span></span>
<span><span class="co">#&gt; [67] R6_2.5.1                textshaping_0.3.6       rprojroot_2.0.3        </span></span>
<span><span class="co">#&gt; [70] evaluate_0.21           lattice_0.21-8          highr_0.10             </span></span>
<span><span class="co">#&gt; [73] png_0.1-8               memoise_2.0.1           bslib_0.5.0            </span></span>
<span><span class="co">#&gt; [76] Rcpp_1.0.11             nlme_3.1-162            mgcv_1.8-42            </span></span>
<span><span class="co">#&gt; [79] xfun_0.39               fs_1.6.2                pkgconfig_2.0.3</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Dongyuan Song, Qingyang Wang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
