<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="extract_para generates parameter matrices which determine each cell's distribution"><title>Extract the parameters of each cell's distribution — extract_para • scDesign3</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Extract the parameters of each cell's distribution — extract_para"><meta property="og:description" content="extract_para generates parameter matrices which determine each cell's distribution"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scDesign3</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/scDesign3.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/scDesign3-introduction-vignette.html">scDesign3 Introduction</a>
    <a class="dropdown-item" href="../articles/scDesign3-marginal-vignette.html">scDesign3 marginal distribution for genes</a>
    <a class="dropdown-item" href="../articles/scDesign3-copulaCompare-vignette.html">Compare gaussian copula and vine copula</a>
    <a class="dropdown-item" href="../articles/scDesign3-parallelization-vignette.html">Parallelization in scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-librarySize-vignette.html">Simulate datasets with cell library size</a>
    <a class="dropdown-item" href="../articles/scDesign3-multipleLineages-vignette.html">Simulate datasets with multiple lineages</a>
    <a class="dropdown-item" href="../articles/scDesign3-conditionEffect-vignette.html">Simulate datasets with condition effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-batchEffect-vignette.html">Simulate datasets with batch effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-spatial-vignette.html">Simulate spatial transcriptomic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-spatial-deconvolution.html">Simulate spot-resolution spatial data for cell-type deconvolution</a>
    <a class="dropdown-item" href="../articles/scDesign3-scATACseq-vignette.html">Simulate single-cell ATAC-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-CITEseq-vignette.html">Simulate CITE-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-multiomics-vignette.html">Simulate multi-omics data from single-omic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-clusterGOF-vignette.html">Evaluate clustering goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-pseudotimeGOF-vignette.html">Evaluate pseudotime goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-DEanalysis-vignette.html">Benchmarking DE Analysis with scDesign3</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/SONGDONGYUAN1994/scDesign3/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Extract the parameters of each cell's distribution</h1>
      <small class="dont-index">Source: <a href="https://github.com/SONGDONGYUAN1994/scDesign3/blob/HEAD/R/extract_para.R" class="external-link"><code>R/extract_para.R</code></a></small>
      <div class="d-none name"><code>extract_para.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>extract_para</code> generates parameter matrices which determine each cell's distribution</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">extract_para</span><span class="op">(</span></span>
<span>  <span class="va">sce</span>,</span>
<span>  assay_use <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  <span class="va">marginal_list</span>,</span>
<span>  <span class="va">n_cores</span>,</span>
<span>  <span class="va">family_use</span>,</span>
<span>  <span class="va">new_covariate</span>,</span>
<span>  parallelization <span class="op">=</span> <span class="st">"mcmapply"</span>,</span>
<span>  BPPARAM <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">data</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>sce</dt>
<dd><p>A <code>SingleCellExperiment</code> object.</p></dd>


<dt>assay_use</dt>
<dd><p>A string which indicates the assay you will use in the sce. Default is 'counts'.</p></dd>


<dt>marginal_list</dt>
<dd><p>A list of fitted regression models from <code><a href="fit_marginal.html">fit_marginal</a></code> for each gene in sce.</p></dd>


<dt>n_cores</dt>
<dd><p>An integer. The number of cores to use.</p></dd>


<dt>family_use</dt>
<dd><p>A string of the marginal distribution.
Must be one of 'poisson', 'nb', 'zip', 'zinb' or 'gaussian', which represent 'poisson distribution',
'negative binomial distribution', 'zero-inflated poisson distribution', 'zero-inflated negative binomial distribution',
and 'gaussian distribution' respectively.</p></dd>


<dt>new_covariate</dt>
<dd><p>A data.frame which contains covariates of targeted simulated data from  <code><a href="construct_data.html">construct_data</a></code> and the
correlation group assignment for each cell in the column 'corr_group'.</p></dd>


<dt>parallelization</dt>
<dd><p>A string indicating the specific parallelization function to use.
Must be one of 'mcmapply', 'bpmapply', or 'pbmcmapply', which corresponds to the parallelization function in the package
<code>parallel</code>,<code>BiocParallel</code>, and <code>pbmcapply</code> respectively. The default value is 'mcmapply'.</p></dd>


<dt>BPPARAM</dt>
<dd><p>A <code>MulticoreParam</code> object or NULL. When the parameter parallelization = 'mcmapply' or 'pbmcmapply',
this parameter must be NULL. When the parameter parallelization = 'bpmapply',  this parameter must be one of the
<code>MulticoreParam</code> object offered by the package 'BiocParallel. The default value is NULL.</p></dd>


<dt>data</dt>
<dd><p>A dataframe which is used when fitting the gamlss model</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A list with the components:</p><dl><dt><code>mean_mat</code></dt>
<dd><p>A cell by feature matrix of the mean parameter.</p></dd>

  <dt><code>sigma_mat</code></dt>
<dd><p>A cell by feature matrix of the sigma parameter (for Gaussian, the variance; for NB, the dispersion.).</p></dd>

  <dt><code>zero_mat</code></dt>
<dd><p>A cell by feature matrix of the zero-inflation parameter (only non-zero for ZIP and ZINB).</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The function takes the new covariate (if use) from <code><a href="construct_data.html">construct_data</a></code> and
marginal models from <code><a href="fit_marginal.html">fit_marginal</a></code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="va">my_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="construct_data.html">construct_data</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  sce <span class="op">=</span> <span class="va">example_sce</span>,</span></span>
<span class="r-in"><span>  assay_use <span class="op">=</span> <span class="st">"counts"</span>,</span></span>
<span class="r-in"><span>  celltype <span class="op">=</span> <span class="st">"cell_type"</span>,</span></span>
<span class="r-in"><span>  pseudotime <span class="op">=</span> <span class="st">"pseudotime"</span>,</span></span>
<span class="r-in"><span>  spatial <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>  other_covariates <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>  corr_by <span class="op">=</span> <span class="st">"1"</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="va">my_marginal</span> <span class="op">&lt;-</span> <span class="fu"><a href="fit_marginal.html">fit_marginal</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  data <span class="op">=</span> <span class="va">my_data</span>,</span></span>
<span class="r-in"><span>  mu_formula <span class="op">=</span> <span class="st">"s(pseudotime, bs = 'cr', k = 10)"</span>,</span></span>
<span class="r-in"><span>  sigma_formula <span class="op">=</span> <span class="st">"1"</span>,</span></span>
<span class="r-in"><span>  family_use <span class="op">=</span> <span class="st">"nb"</span>,</span></span>
<span class="r-in"><span>  n_cores <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>  usebam <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="va">my_copula</span> <span class="op">&lt;-</span> <span class="fu"><a href="fit_copula.html">fit_copula</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  sce <span class="op">=</span> <span class="va">example_sce</span>,</span></span>
<span class="r-in"><span>  assay_use <span class="op">=</span> <span class="st">"counts"</span>,</span></span>
<span class="r-in"><span>  marginal_list <span class="op">=</span> <span class="va">my_marginal</span>,</span></span>
<span class="r-in"><span>  family_use <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"nb"</span>, <span class="fl">5</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"zip"</span>, <span class="fl">5</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  copula <span class="op">=</span> <span class="st">"vine"</span>,</span></span>
<span class="r-in"><span>  n_cores <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>  input_data <span class="op">=</span> <span class="va">my_data</span><span class="op">$</span><span class="va">dat</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Convert Residuals to Uniform</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting End</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Copula group 1 starts</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Vine Copula Estimation Starts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.1801219 secs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Vine Copula Estimation Ends</span>
<span class="r-in"><span>  <span class="va">my_para</span> <span class="op">&lt;-</span> <span class="fu">extract_para</span><span class="op">(</span></span></span>
<span class="r-in"><span>    sce <span class="op">=</span> <span class="va">example_sce</span>,</span></span>
<span class="r-in"><span>    marginal_list <span class="op">=</span> <span class="va">my_marginal</span>,</span></span>
<span class="r-in"><span>    n_cores <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>    family_use <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"nb"</span>, <span class="fl">5</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"zip"</span>, <span class="fl">5</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    new_covariate <span class="op">=</span> <span class="va">my_data</span><span class="op">$</span><span class="va">new_covariate</span>,</span></span>
<span class="r-in"><span>    data <span class="op">=</span> <span class="va">my_data</span><span class="op">$</span><span class="va">dat</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Dongyuan Song, Qingyang Wang, Chenxin Jiang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

