<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="scdesign3 takes the input data, fits the model and"><title>The wrapper for the whole scDesign3 pipeline — scdesign3 • scDesign3</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="The wrapper for the whole scDesign3 pipeline — scdesign3"><meta property="og:description" content="scdesign3 takes the input data, fits the model and"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scDesign3</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/scDesign3.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/scDesign3-introduction-vignette.html">scDesign3 Introduction</a>
    <a class="dropdown-item" href="../articles/scDesign3-marginal-vignette.html">scDesign3 marginal distribution for genes</a>
    <a class="dropdown-item" href="../articles/scDesign3-copulaCompare-vignette.html">Compare gaussian copula and vine copula</a>
    <a class="dropdown-item" href="../articles/scDesign3-parallelization-vignette.html">Parallelization in scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-librarySize-vignette.html">Simulate datasets with cell library size</a>
    <a class="dropdown-item" href="../articles/scDesign3-multipleLineages-vignette.html">Simulate datasets with multiple lineages</a>
    <a class="dropdown-item" href="../articles/scDesign3-conditionEffect-vignette.html">Simulate datasets with condition effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-batchEffect-vignette.html">Simulate datasets with batch effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-spatial-vignette.html">Simulate spatial transcriptomic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-spatial-deconvolution.html">Simulate spot-resolution spatial data for cell-type deconvolution</a>
    <a class="dropdown-item" href="../articles/scDesign3-scATACseq-vignette.html">Simulate single-cell ATAC-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-CITEseq-vignette.html">Simulate CITE-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-multiomics-vignette.html">Simulate multi-omics data from single-omic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-clusterGOF-vignette.html">Evaluate clustering goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-pseudotimeGOF-vignette.html">Evaluate pseudotime goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-DEanalysis-vignette.html">Benchmarking DE Analysis with scDesign3</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/SONGDONGYUAN1994/scDesign3/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>The wrapper for the whole scDesign3 pipeline</h1>
      <small class="dont-index">Source: <a href="https://github.com/SONGDONGYUAN1994/scDesign3/blob/HEAD/R/scdesign3.R" class="external-link"><code>R/scdesign3.R</code></a></small>
      <div class="d-none name"><code>scdesign3.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>scdesign3</code> takes the input data, fits the model and</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">scdesign3</span><span class="op">(</span></span>
<span>  <span class="va">sce</span>,</span>
<span>  assay_use <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  <span class="va">celltype</span>,</span>
<span>  pseudotime <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  spatial <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">other_covariates</span>,</span>
<span>  ncell <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,</span>
<span>  <span class="va">mu_formula</span>,</span>
<span>  sigma_formula <span class="op">=</span> <span class="st">"1"</span>,</span>
<span>  family_use <span class="op">=</span> <span class="st">"nb"</span>,</span>
<span>  n_cores <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  usebam <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  edf_flexible <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">corr_formula</span>,</span>
<span>  empirical_quantile <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  copula <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>  if_sparse <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  fastmvn <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  DT <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pseudo_obs <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  family_set <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gauss"</span>, <span class="st">"indep"</span><span class="op">)</span>,</span>
<span>  important_feature <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>  nonnegative <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  nonzerovar <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  return_model <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  simplify <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  parallelization <span class="op">=</span> <span class="st">"mcmapply"</span>,</span>
<span>  BPPARAM <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  trace <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>sce</dt>
<dd><p>A <code>SingleCellExperiment</code> object.</p></dd>


<dt>assay_use</dt>
<dd><p>A string which indicates the assay you will use in the sce. Default is 'counts'.</p></dd>


<dt>celltype</dt>
<dd><p>A string of the name of cell type variable in the <code>colData</code> of the sce. Default is 'cell_type'.</p></dd>


<dt>pseudotime</dt>
<dd><p>A string or a string vector of the name of pseudotime and (if exist)
multiple lineages. Default is NULL.</p></dd>


<dt>spatial</dt>
<dd><p>A length two string vector of the names of spatial coordinates. Default is NULL.</p></dd>


<dt>other_covariates</dt>
<dd><p>A string or a string vector of the other covariates you want to include in the data.</p></dd>


<dt>ncell</dt>
<dd><p>The number of cell you want to simulate. Default is <code>dim(sce)[2]</code> (the same number as the input data).</p></dd>


<dt>mu_formula</dt>
<dd><p>A string of the mu parameter formula</p></dd>


<dt>sigma_formula</dt>
<dd><p>A string of the sigma parameter formula</p></dd>


<dt>family_use</dt>
<dd><p>A string of the marginal distribution.
Must be one of 'poisson', 'nb', 'zip', 'zinb' or 'gaussian'.</p></dd>


<dt>n_cores</dt>
<dd><p>An integer. The number of cores to use.</p></dd>


<dt>usebam</dt>
<dd><p>A logic variable. If use <code><a href="https://rdrr.io/pkg/mgcv/man/bam.html" class="external-link">bam</a></code> for acceleration in marginal fitting.</p></dd>


<dt>edf_flexible</dt>
<dd><p>A logic variable. It is used for accelerating for spatial model if k is large in 'mu_formula'. Default is FALSE.</p></dd>


<dt>corr_formula</dt>
<dd><p>A string of the correlation structure.</p></dd>


<dt>empirical_quantile</dt>
<dd><p>Please only use it if you clearly know what will happen! A logic variable. If TRUE, DO NOT fit the copula and use the EMPIRICAL CDF values of the original data; it will make the simulated data fixed (no randomness). Default is FALSE. Only works if ncell is the same as your original data.</p></dd>


<dt>copula</dt>
<dd><p>A string of the copula choice. Must be one of 'gaussian' or 'vine'. Default is 'gaussian'. Note that vine copula may have better modeling of high-dimensions, but can be very slow when features are &gt;1000.</p></dd>


<dt>if_sparse</dt>
<dd><p>A logic variable. Only works for Gaussian copula (<code>family_set = "gaussian"</code>). If TRUE, a thresholding strategy will make the corr matrix sparse.</p></dd>


<dt>fastmvn</dt>
<dd><p>An logical variable. If TRUE, the sampling of multivariate Gaussian is done by <code>mvnfast</code>, otherwise by <code>mvtnorm</code>. Default is FALSE. It only matters for Gaussian copula.</p></dd>


<dt>DT</dt>
<dd><p>A logic variable. If TRUE, perform the distributional transformation
to make the discrete data 'continuous'. This is useful for discrete distributions (e.g., Poisson, NB).
Default is TRUE. Note that for continuous data (e.g., Gaussian), DT does not make sense and should be set as FALSE.</p></dd>


<dt>pseudo_obs</dt>
<dd><p>A logic variable. If TRUE, use the empirical quantiles instead of theoretical quantiles for fitting copula.
Default is FALSE.</p></dd>


<dt>family_set</dt>
<dd><p>A string or a string vector of the bivariate copula families. Default is c("gauss", "indep"). For more information please check package <code>rvinecoplib</code>.</p></dd>


<dt>important_feature</dt>
<dd><p>A numeric value or vector which indicates whether a gene will be used in correlation estimation or not. If this is a numeric value, then
gene with zero proportion greater than this value will be excluded form gene-gene correlation estimation. If this is a vector, then this should
be a logical vector with length equal to the number of genes in <code>sce</code>. <code>TRUE</code> in the logical vector means the corresponding gene will be included in
gene-gene correlation estimation and <code>FALSE</code> in the logical vector means the corresponding gene will be excluded from the gene-gene correlation estimation.
The default value for is "all" (a special string which means no filtering).</p></dd>


<dt>nonnegative</dt>
<dd><p>A logical variable. If TRUE, values &lt; 0 in the synthetic data will be converted to 0. Default is TRUE (since the expression matrix is nonnegative).</p></dd>


<dt>nonzerovar</dt>
<dd><p>A logical variable. If TRUE, for any gene with zero variance, a cell will be replaced with 1. This is designed for avoiding potential errors, for example, PCA. Default is FALSE.</p></dd>


<dt>return_model</dt>
<dd><p>A logic variable. If TRUE, the marginal models and copula models will be returned. Default is FALSE.</p></dd>


<dt>simplify</dt>
<dd><p>A logic variable. If TRUE, the fitted regression model will only keep the essential contains for <code>predict</code>, otherwise the fitted models can be VERY large. Default is FALSE.</p></dd>


<dt>parallelization</dt>
<dd><p>A string indicating the specific parallelization function to use.
Must be one of 'mcmapply', 'bpmapply', or 'pbmcmapply', which corresponds to the parallelization function in the package
<code>parallel</code>,<code>BiocParallel</code>, and <code>pbmcapply</code> respectively. The default value is 'mcmapply'.</p></dd>


<dt>BPPARAM</dt>
<dd><p>A <code>MulticoreParam</code> object or NULL. When the parameter parallelization = 'mcmapply' or 'pbmcmapply',
this parameter must be NULL. When the parameter parallelization = 'bpmapply',  this parameter must be one of the
<code>MulticoreParam</code> object offered by the package 'BiocParallel. The default value is NULL.</p></dd>


<dt>trace</dt>
<dd><p>A logic variable. If TRUE, the warning/error log and runtime for gam/gamlss
will be returned, FALSE otherwise. Default is FALSE.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A list with the components:</p><dl><dt><code>new_count</code></dt>
<dd><p>A matrix of the new simulated count (expression) matrix.</p></dd>

  <dt><code>new_covariate</code></dt>
<dd><p>A data.frame of the new covariate matrix.</p></dd>

  <dt><code>model_aic</code></dt>
<dd><p>The model AIC.</p></dd>

  <dt><code>marginal_list</code></dt>
<dd><p>A list of marginal regression models if return_model = TRUE.</p></dd>

  <dt><code>corr_list</code></dt>
<dd><p>A list of correlation models (conditional copulas) if return_model = TRUE.</p></dd>


</dl></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">my_simu</span> <span class="op">&lt;-</span> <span class="fu">scdesign3</span><span class="op">(</span></span></span>
<span class="r-in"><span>sce <span class="op">=</span> <span class="va">example_sce</span>,</span></span>
<span class="r-in"><span>assay_use <span class="op">=</span> <span class="st">"counts"</span>,</span></span>
<span class="r-in"><span>celltype <span class="op">=</span> <span class="st">"cell_type"</span>,</span></span>
<span class="r-in"><span>pseudotime <span class="op">=</span> <span class="st">"pseudotime"</span>,</span></span>
<span class="r-in"><span>spatial <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>other_covariates <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>mu_formula <span class="op">=</span> <span class="st">"s(pseudotime, bs = 'cr', k = 10)"</span>,</span></span>
<span class="r-in"><span>sigma_formula <span class="op">=</span> <span class="st">"1"</span>,</span></span>
<span class="r-in"><span>family_use <span class="op">=</span> <span class="st">"nb"</span>,</span></span>
<span class="r-in"><span>n_cores <span class="op">=</span> <span class="fl">2</span>,</span></span>
<span class="r-in"><span>usebam <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>edf_flexible <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>corr_formula <span class="op">=</span> <span class="st">"pseudotime"</span>,</span></span>
<span class="r-in"><span>copula <span class="op">=</span> <span class="st">"gaussian"</span>,</span></span>
<span class="r-in"><span>if_sparse <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>DT <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>pseudo_obs <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>ncell <span class="op">=</span> <span class="fl">1000</span>,</span></span>
<span class="r-in"><span>return_model <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Input Data Construction Start</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> fitting ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |======                                                                |   9%  |                                                                              |=============                                                         |  18%  |                                                                              |===================                                                   |  27%  |                                                                              |=========================                                             |  36%  |                                                                              |================================                                      |  45%  |                                                                              |======================================                                |  55%  |                                                                              |=============================================                         |  64%  |                                                                              |===================================================                   |  73%  |                                                                              |=========================================================             |  82%  |                                                                              |================================================================      |  91%  |                                                                              |======================================================================| 100%</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Input Data Construction End</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Start Marginal Fitting</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Marginal Fitting End</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Start Copula Fitting</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Convert Residuals to Multivariate Gaussian</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting End</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Copula group 1 starts</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Copula group 2 starts</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Copula group 3 starts</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Copula group 4 starts</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Copula Fitting End</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Start Parameter Extraction</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Parameter</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extraction End</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Start Generate New Data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Use Copula to sample a multivariate quantile matrix</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sample Copula group 1 starts</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sample Copula group 2 starts</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sample Copula group 3 starts</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Sample Copula group 4 starts</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> New Data Generating End</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Dongyuan Song, Qingyang Wang, Chenxin Jiang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

